src/components/NewCommentForm.tsx

```
@@ -1,8 +1,90 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import PropTypes from 'prop-types';
+
+type Props = {
+  onSubmit: (comment: Omit<Comment, 'id'>) => Promise<void>;
+  selectedPost?: Post;
+};
+
+type FormState = {
+  name: string;
+  email: string;
+  body: string;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onSubmit, selectedPost }) => {
+  const [form, setForm] = useState<FormState>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [errors, setErrors] = useState<FormState>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  function handleChange<K extends keyof FormState>(key: K, value: string) {
+    setForm(prev => ({ ...prev, [key]: value }));
+    setErrors(prev => ({ ...prev, [key]: '' }));
+  }
+
+  function clearForm() {
+    setForm({ name: '', email: '', body: '' });
+    setErrors({ name: '', email: '', body: '' });
+  }
+
+  function resetBody() {
+    setForm(prev => ({ ...prev, body: '' }));
+    setErrors(prev => ({ ...prev, body: '' }));
+  }
+
+  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    if (!selectedPost?.id) {
+      return;
+    }
+
+    const newErrors: FormState = {
+      name: form.name.trim() ? '' : 'Name is required',
+      email: form.email.trim() ? '' : 'Email is required',
+      body: form.body.trim() ? '' : 'Comment is required',
+    };
+
+    setErrors(newErrors);
+
+    if (Object.values(newErrors).some(Boolean)) {
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+
+      const payload = {
+        name: form.name.trim(),
+        email: form.email.trim(),
+        body: form.body.trim(),
+        postId: selectedPost.id,
+      };
+
+      await onSubmit(payload);
+      resetBody();
+    } catch (error) {
+      throw error;
```

This violates the previous review / checklist instruction: "Replace empty catch blocks with at least `console.error(error)` and, where appropriate, set error state." The catch block currently re-throws the error (`throw error;`) which will bubble the error instead of recording/logging it and providing the user a way to retry. Please update the catch block to at least `console.error(error)` and consider setting an error state (or calling a passed error handler) so the UI can present an error notification and allow retry. Avoid rethrowing here so you don't cause uncaught promise rejections in the UI flow.